### PR TITLE
bpo-30184: Add tests for invalid use of PyArg_ParseTupleAndKeywords.

### PR DESCRIPTION
**Pull request by serhiy-storchaka**

### CODE REVIEW

File: Lib/test/test_capi.py [hunk 1]
```diff
@@ -517,40 +516,64 @@ def test_skipitem(self):
...
+        self.assertRaises(SystemError, _testcapi.parse_tuple_and_keywords,
+                          (), {'a': 1, 'b': 2}, '$O|O', ['a', 'b'])
+        self.assertRaises(SystemError, _testcapi.parse_tuple_and_keywords,
+                          (1,), {}, '|O', ['a', 'b'])
+        self.assertRaises(SystemError, _testcapi.parse_tuple_and_keywords,
+                          (1,), {}, '|OO', ['a'])
+        self.assertRaises(SystemError, _testcapi.parse_tuple_and_keywords,
+                          (), {}, '|$O', [''])
+        self.assertRaises(SystemError, _testcapi.parse_tuple_and_keywords,
+                          (), {}, '|OO', ['a', ''])
```
Comment (louisom) [inline Lib/test/test_capi.py:551]: May add test case for misuse of `;` and `:`

```python
        self.assertRaises(SystemError, _testcapi.parse_tuple_and_keywords,
                          (1,), {}, ':O', ['a'])
        self.assertRaises(SystemError, _testcapi.parse_tuple_and_keywords,
                          (1,), {}, ';O', ['a'])
```

File: Lib/test/test_capi.py [hunk 2]
```diff
@@ -517,40 +516,64 @@ def test_skipitem(self):
...
         self.assertRaises(ValueError, _testcapi.parse_tuple_and_keywords,
-                          (), {}, b'', [''] * 42)
+                          (), {}, '', [''] * 42)
         self.assertRaises(ValueError, _testcapi.parse_tuple_and_keywords,
-                          (), {}, b'', [42])
+                          (), {}, '', [42])
+
+    def test_bad_use(self):
+        # Test handling invalid format and keywords in
+        # PyArg_ParseTupleAndKeywords()
```
Comment (louisom) [inline Lib/test/test_capi.py:531]: Can this use the method below `parser = _testcapi.parse_tuple_and_keywords` to reduce the length?

**Review by louisom — CHANGES_REQUESTED** [review]
May add some test case

File: Lib/test/test_capi.py (hunk 2 above)
Comment (serhiy-storchaka) [reply Lib/test/test_capi.py:531]: This doesn't matter, in any case the line is wrapped.

File: Lib/test/test_capi.py (hunk 1 above)
Comment (serhiy-storchaka) [reply Lib/test/test_capi.py:551]: These tests don't trigger any error that is not tested.

**Pull request merged**